modules/afsocket/afsocket-dest.c

```diff
@@ -207,6 +233,46 @@ afsocket_dd_start_reconnect_timer(AFSocketDestDriver *self)
   iv_timer_register(&self->reconnect_timer);
 }
 
+// Returns TRUE in case of success, or there is nothing to do. FALSE will indicate an error.
+static gboolean
+_afsocket_dd_rename_connection_persist_name(AFSocketDestDriver *self, PersistState *state)
+{
+  const gchar *old_name, *new_name;
+  PersistEntryHandle old_handle, new_handle;
+  gsize size;
+  guint8 old_version, new_version;
+  gpointer old_value, new_value;
+
+  // Make sure there is no entry with the new name
+  new_name = afsocket_dd_format_connections_name(self);
+  new_handle = persist_state_lookup_entry(state, new_name, &size, &new_version);
+  if (new_handle)
+    return TRUE;
+
+  // Find the old entry
+  old_name = afsocket_dd_format_legacy_connection_name(self);
+  old_handle = persist_state_lookup_entry(state, old_name, &size, &old_version);
+  if (!old_handle)
+    return TRUE;
```

Comment:
The first 2 sections can be extracted, so you could write something like this in `afsocket_dd_construct_transport`:
```
if (_need_to_rename_legacy_persist_key(self, state) &&
    !afsocket_dd_rename_connection_persist_name(self, cfg->state))
{
  return FALSE;
}
```
or `_legacy_persist_entry_exists()` if you want to remove the legacy entry after the rename (or anything else :)).

This way, only the third section remains in `rename_connection_persist_name`, which is the real "rename" part.

Reply:
It's hard to implement it this way, marking as resolved.